src/lcm/model.py

```diff
@@ -62,19 +62,21 @@ def __init__(
             enable_jit: Whether to jit the functions of the internal regime.
 
         """
-        _validate_model_inputs(
-            n_periods=n_periods,
-            regime=regime,
-        )
-
         self.n_periods = n_periods
         self.description = description
         self.enable_jit = enable_jit
-        self.regime = regime
-
-        self.internal_regime = process_regime(
-            regime=regime, n_periods=n_periods, enable_jit=enable_jit
-        )
+        self.regimes = {}
+        self.internal_regimes = {}
+
+        for regime in regimes:
+            _validate_model_inputs(
+                n_periods=n_periods,
+                regime=regime,
+            )
+            self.regimes[regime.name] = regime
+            self.internal_regimes[regime.name] = process_regime(
+                regime=regime, n_periods=n_periods, enable_jit=enable_jit
```

Comment:
Don't we have to update the regime processing to include all regimes? I thought that we need to do the processing of certain regime components first, as we then need to pass the processed components of all regimes into each regime? I.e., the Q-and-F function of a regime depends on all of the other regimes?